### PR TITLE
impr: prefer @SuperBuilder to @Builder for objects extending a class [FPLA-NA]

### PR DESCRIPTION
**Pull request by Gogetter**
### Change description ###

prefer @SuperBuilder to @Builder to avoid parameters of more than 7-args

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: service/src/main/java/uk/gov/hmcts/reform/fpl/model/docmosis/DocmosisCaseManagementOrder.java [hunk 1]
```diff
@@ -1,47 +1,21 @@
...
-            children, respondentsProvided, applicantName, hearingBooking, directions, draftbackground, courtseal,
-            crest);
-        this.representatives = representatives;
-        this.schedule = schedule;
-        this.scheduleProvided = scheduleProvided;
-        this.recitals = recitals;
-        this.recitalsProvided = recitalsProvided;
-    }
+    private final DocmosisJudgeAndLegalAdvisor allocatedJudgeAndLegalAdvisor;
+    private final int numberOfChildren;
```
Comment (lewisbirks) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/model/docmosis/DocmosisCaseManagementOrder.java:20]: I don't think that this field is needed, isn't used in the document

Comment (Gogetter) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/model/docmosis/DocmosisCaseManagementOrder.java:20]: thanks @lewisbirks

Comment (lewisbirks) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/model/docmosis/DocmosisCaseManagementOrder.java:20]: I'm guessing the other new field is from your other story? Might be best not to have it in this one as it is strange that you are adding fields when just replacing the constructor.

**Comment by muyinatech** [comment]
All code review comments addressed and pushed. Please can you review when convenient.

**Pull request merged**